src/maxminddb_stubs.c

```diff
@@ -217,19 +217,23 @@ CAMLprim value mmdb_ml_lookup_path(value ip, value query_list, value mmdb)
 			       (const char *const *const)query);
   check_status(status);
   check_data(entry_data);
-  free(result);
-  for (int i = 0; i < copy_count; free(query[i]), i++);
-  free(query);
+  caml_stat_free(result);
+  for (int i = 0; i < copy_count; caml_stat_free(query[i]), i++);
+  caml_stat_free(query);
   query_r = caml_alloc(2, 0);
   as_mmdb = NULL;
   switch (entry_data.type) {
   case MMDB_DATA_TYPE_BYTES:
     clean_result = caml_stat_alloc(entry_data.data_size + 1);
     memcpy(clean_result, entry_data.bytes, entry_data.data_size);
+    caml_clean_result = caml_copy_string(clean_result);
+    caml_stat_free(clean_result);
     goto string_finish;
 
   case MMDB_DATA_TYPE_UTF8_STRING:
     clean_result = strndup(entry_data.utf8_string, entry_data.data_size);
+    caml_clean_result = caml_copy_string(clean_result);
+    free(clean_result);
```

Comment:
I'm just having a look at the various findings of @cakeplus out of curiosity, but I was a bit surprised by the deduplication of the `string_finish` logic and in fact the code of these two cases, BYTES and UTF8_STRING: it seems that the creation of the `caml_clean_result` involves two copies in both cases, is there a reason for not calling `caml_copy_string` directly on the `entry_data` string?